arangod/IResearch/IResearchInvertedIndexMeta.cpp

```diff
@@ -831,29 +804,28 @@ std::string_view InvertedIndexField::attributeString() const {
 
 std::string_view InvertedIndexField::path() const noexcept { return _path; }
 
-bool InvertedIndexField::namesMatch(
+bool InvertedIndexField::operator==(
     InvertedIndexField const& other) const noexcept {
-  return analyzerName() == other.analyzerName() &&
-         basics::AttributeName::namesMatch(_attribute, other._attribute);
-}
-
-bool InvertedIndexField::isIdentical(
-    std::vector<basics::AttributeName> const& path,
-    irs::string_ref analyzerName) const noexcept {
-  if (_analyzers.front()._shortName == analyzerName &&
-      path.size() == _attribute.size()) {
-    auto it = path.begin();
-    auto atr = _attribute.begin();
-    while (it != path.end() && atr != _attribute.end()) {
-      if (it->name != atr->name || it->shouldExpand != atr->shouldExpand) {
-        return false;
+  if (!(analyzerName() == other.analyzerName() &&
+        basics::AttributeName::namesMatch(_attribute, other._attribute) &&
```

Comment:
I think you can sort condition to faster check. First of bool, int == int, then string == string, then more compilcated